### DISCUSSION TITLE
[@types/ramda] curried version of `prop` does not support undefined value

### DISCUSSION THREAD

**Discussion by anion155** [DefinitelyTyped/DefinitelyTyped#59867] [Issues with a @types package]
```
const value: { a: number } | undefined = { a: 5 };
// typescript
pipe(
  defaultTo({}), // ts will fail without it
  prop('a')
)(value)
// javascript
pipe(prop('a'))(undefined) // => undefined
```

[a comment by an automated account is not shown]

**Comment by adispring** [comment]
> ```
> const value: { a: number } | undefined = { a: 5 };
> // typescript
> pipe(
>   defaultTo({}), // ts will fail without it
>   prop('a')
> )(value)
> // javascript
> pipe(prop('a'))(undefined) // => undefined
> ```

Maybe this TS's behavior of `prop` is more safer?

**Reply by anion155** [reply]
I think it should be pretty safe, to declare additional function declaration, which would accept `Record<P, T> | undefined` and return `T | undefined`

**Comment by essenmitsosse** [comment]
Since this actually seems to be the behaviour of `prop` we might as well just add an overload that is something like 

```typescript
function(key: string, value: undefined): undefined
function(key: string): (value: undefined) => undefined
```

**Reply by adispring** [reply]
Then `prop` should also process `null`.

**Reply by essenmitsosse** [reply]
```typescript
function(key: string, value: undefined | null): undefined
function(key: string): (value: undefined | null) => undefined
```

**Reply by adispring** [reply]
There are many other apis that can handle `undefined` and `null`, should all of them add overloads to handle this case?
